### PR TITLE
[AIX][CMake] Disable `openmp` as LLVM_ENABLE_PROJECTS 

### PR DESCRIPTION
**Pull request by DanielCChen**
in favor of LLVM_ENABLE_RUNTIMES

### CODE REVIEW

File: llvm/CMakeLists.txt [hunk 1]
```diff
@@ -117,6 +117,13 @@ endif()
 # This allows an easy way of setting up a build directory for llvm and another
 # one for llvm+clang+... using the same sources.
 set(LLVM_ALL_PROJECTS "bolt;clang;clang-tools-extra;compiler-rt;cross-project-tests;libc;libclc;lld;lldb;mlir;openmp;polly;pstl")
+if (${CMAKE_SYSTEM_NAME} MATCHES "AIX")
+  # Disallow 'openmp' as a LLVM PROJECT on AIX as the supported way is to use
+  # LLVM_ENABLE_RUNTIMES.
+  list(REMOVE_ITEM LLVM_ALL_PROJECTS openmp)
+endif()
+
+# The flang project is not yet part of "all" projects (see C++ requirements)
```
Comment (daltenty) [inline llvm/CMakeLists.txt:126]: nit: this is a dup. I think this is a duplicate/copy-paste error:
```suggestion
```

**Review by daltenty — APPROVED** [review]
LGTM with minor nit

File: llvm/CMakeLists.txt (hunk 1 above)
Comment (DanielCChen) [reply llvm/CMakeLists.txt:126]: Thanks for the review. The dup line is removed.

**Comment by shiltian** [comment]
What is the reason? Usually `libomp` doesn't need to be in runtimes.

**Comment by DanielCChen** [comment]
> What is the reason? Usually `libomp` doesn't need to be in runtimes.

On AIX, building `openmp` as a LLVM project never worked. It failed due to some non-standard code in AIX header files. Even though we believe building `libomp` should always be part of the runtimes, we only disable it for AIX.

**Pull request merged**